todo-react/src/components/Card.tsx

```diff
@@ -0,0 +1,167 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  MdClose,
+  MdDone,
+  MdEdit,
+  MdOutlineCheckCircleOutline,
+  MdHighlightOff,
+} from "react-icons/md";
+import styled from "styled-components";
+import { Todo } from "todo";
+
+const Wrapper = styled.li<{ done: boolean }>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 1rem;
+  display: flex;
+  background-color: ${({ done }) => (done ? "#2c3e50" : "#fff")};
+  color: ${({ done }) => (done ? "#fff" : "#2c3e50")};
+  border-radius: 0.5rem;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  margin-bottom: 1rem;
+  transition: all 0.2s ease-in-out;
+`;
+
+const TitleContainer = styled.div``;
+
+const CreatedAt = styled.span`
+  font-size: 0.8rem;
+  color: inherit;
+  opacity: 0.5;
+`;
+
+const ControlContainer = styled.div<{ done: boolean }>`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  gap: 0.5rem;
+  svg {
+    transition: all 0.2s ease-in-out;
+    color: inherit;
+    cursor: pointer;
+    font-size: 2.5rem;
+    padding: 0.5rem;
+    border-radius: 9999px;
+    :hover {
+      background-color: ${({ done }) => (done ? "#fff" : "#2c3e50")};
+      color: ${({ done }) => (done ? "#2c3e50" : "#fff")};
+    }
+    :last-child:hover {
+      background-color: #e74c3c;
+      color: #fff;
+    }
+  }
+`;
+
+const NewTodoForm = styled.form``;
+
+const NewTodoInput = styled.input.attrs({
+  type: "text",
+  required: true,
+  maxLength: 20,
+})`
+  all: unset;
+`;
+
+const Title = styled.h3``;
+
+interface CardProps extends Todo {
+  handleEditTodo: (id: number, newTitle: string) => void;
+  handleDeleteTodo: (id: number) => void;
+  handleCompleteTodo: (id: number) => void;
+}
+
+const Card = ({
+  title,
+  createdAt,
+  id,
+  completed,
+  handleEditTodo,
+  handleDeleteTodo,
+  handleCompleteTodo,
+}: CardProps) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const editTitle = ({
+    currentTarget: { value },
+  }: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(value);
+  };
+
+  const toggleEdit = useCallback(() => {
+    if (newTitle !== title) setNewTitle(title);
+    setIsEditing(!isEditing);
+  }, [isEditing, newTitle, title]);
+
+  const submitEditedTitle = useCallback(
+    (e?: React.FormEvent<HTMLFormElement>) => {
+      e?.preventDefault();
+      handleEditTodo(id, newTitle);
+      toggleEdit();
+    },
+    [handleEditTodo, id, newTitle, toggleEdit]
+  );
+
+  const todoContent = useMemo(
+    () =>
+      isEditing ? (
+        <NewTodoForm onSubmit={submitEditedTitle}>
+          <NewTodoInput ref={inputRef} value={newTitle} onChange={editTitle} />
```

Comment:
```suggestion
          <NewTodoInput ref={inputRef} value={newTitle} onChange={handleChangeTitle} />
```
`onChange`에 대한 이벤트를 처리하고 있기 때문에 이렇게 표현해주면 더 좋을 것 같아요!